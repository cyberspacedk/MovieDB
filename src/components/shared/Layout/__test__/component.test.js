import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import PageLayout from '../component';

describe('Component: PageLayout', () => {
  const props = {
    title: 'some title',
    removeBox: false,
    loading: false,
    error: false,
    empty: false,
    currentPage: 1,
    totalResults: 0,
    history: {
      push: () => {},
    },
    array: [],
    operations: () => {},
    goToNextPage: () => {},
  };

  it('Match its snapshot. Array provided', () => {
    const nextProps = {
      ...props,
      array: [{ id: 1, title: 'Some title', overview: 'Some description' }],
    };
    const wrapper = shallow(<PageLayout {...nextProps} />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('Match its snapshot. Loading ...', () => {
    const nextProps = {
      ...props,
      loading: true,
    };
    const wrapper = shallow(<PageLayout {...nextProps} />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('Match its snapshot. Error', () => {
    const nextProps = {
      ...props,
      error: true,
    };
    const wrapper = shallow(<PageLayout {...nextProps} />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('Match its snapshot. No title', () => {
    const nextProps = {
      ...props,
      title: '',
    };
    const wrapper = shallow(<PageLayout {...nextProps} />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('Match its snapshot. No remove box', () => {
    const nextProps = {
      ...props,
      removeBox: true,
    };
    const wrapper = shallow(<PageLayout {...nextProps} />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('Match its snapshot. With pagination', () => {
    const nextProps = {
      ...props,
      totalResults: 55,
    };
    const wrapper = shallow(<PageLayout {...nextProps} />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
