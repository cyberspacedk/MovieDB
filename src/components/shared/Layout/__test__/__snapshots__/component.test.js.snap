// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: PageLayout Match its snapshot. Array provided 1`] = `
<Adapter>
  <Adapter>
    <Row
      gutter={0}
    >
      <Col
        offset={2}
        span={20}
      >
        <div
          className="top-margin"
        >
          <Title>
            some title
          </Title>
        </div>
      </Col>
    </Row>
    <div
      className="top-margin"
    >
      <Row
        gutter={16}
        justify="center"
        type="flex"
      >
        <Col
          span={20}
        >
          <CardItemContainer
            history={
              Object {
                "push": [Function],
              }
            }
            item={
              Object {
                "id": 1,
                "overview": "Some description",
                "title": "Some title",
              }
            }
            key="1"
            operations={[Function]}
            removeBox={false}
          />
        </Col>
      </Row>
    </div>
  </Adapter>
  <BackTop
    visibilityHeight={400}
  />
</Adapter>
`;

exports[`Component: PageLayout Match its snapshot. Error 1`] = `
<Adapter>
  <Adapter>
    <Error />
    <Row
      gutter={0}
    >
      <Col
        offset={2}
        span={20}
      >
        <div
          className="top-margin"
        >
          <Title>
            some title
          </Title>
        </div>
      </Col>
    </Row>
    <div
      className="top-margin"
    >
      <Row
        gutter={16}
        justify="center"
        type="flex"
      >
        <Col
          span={20}
        />
      </Row>
    </div>
  </Adapter>
  <BackTop
    visibilityHeight={400}
  />
</Adapter>
`;

exports[`Component: PageLayout Match its snapshot. Loading ... 1`] = `
<Adapter>
  <Adapter>
    <Row
      gutter={0}
    >
      <Col
        offset={2}
        span={20}
      >
        <div
          className="top-margin"
        >
          <Title>
            some title
          </Title>
        </div>
      </Col>
    </Row>
    <Spinner />
  </Adapter>
  <BackTop
    visibilityHeight={400}
  />
</Adapter>
`;

exports[`Component: PageLayout Match its snapshot. No remove box 1`] = `
<Adapter>
  <Adapter>
    <Row
      gutter={0}
    >
      <Col
        offset={2}
        span={20}
      >
        <div
          className="top-margin"
        >
          <Title>
            some title
          </Title>
        </div>
      </Col>
    </Row>
    <div
      className="top-margin"
    >
      <Row
        gutter={16}
        justify="center"
        type="flex"
      >
        <Col
          span={20}
        />
      </Row>
    </div>
  </Adapter>
  <BackTop
    visibilityHeight={400}
  />
</Adapter>
`;

exports[`Component: PageLayout Match its snapshot. No title 1`] = `
<Adapter>
  <Adapter>
    <div
      className="top-margin"
    >
      <Row
        gutter={16}
        justify="center"
        type="flex"
      >
        <Col
          span={20}
        />
      </Row>
    </div>
  </Adapter>
  <BackTop
    visibilityHeight={400}
  />
</Adapter>
`;

exports[`Component: PageLayout Match its snapshot. With pagination 1`] = `
<Adapter>
  <Adapter>
    <Row
      gutter={0}
    >
      <Col
        offset={2}
        span={20}
      >
        <div
          className="top-margin"
        >
          <Title>
            some title
          </Title>
        </div>
      </Col>
    </Row>
    <div
      className="top-margin"
    >
      <Row
        gutter={16}
        justify="center"
        type="flex"
      >
        <Col
          span={20}
        />
      </Row>
      <Row
        gutter={0}
        justify="center"
        type="flex"
      >
        <Col>
          <Pagination
            className="pagination"
            current={1}
            defaultCurrent={1}
            defaultPageSize={20}
            onChange={[Function]}
            total={55}
          />
        </Col>
      </Row>
    </div>
  </Adapter>
  <BackTop
    visibilityHeight={400}
  />
</Adapter>
`;
